lib/crypto/key.go

```diff
@@ -18,17 +18,21 @@ func ReadPub64(suite abstract.Suite, r io.Reader) (abstract.Point, error) {
 // WritePub64 writes a public point to a base64 representation
 func WritePub64(suite abstract.Suite, w io.Writer, point abstract.Point) error {
 	enc := base64.NewEncoder(base64.StdEncoding, w)
-	err := suite.Write(enc, point)
-	enc.Close()
-	return err
+	return write64(suite, enc, point)
+}
+
+func write64(suite abstract.Suite, wc io.WriteCloser, data ...interface{}) error {
+	errW := suite.Write(wc, data)
+	if err := wc.Close(); err != nil && errW == nil {
+		return err
+	}
+	return errW
```

Comment:
Godoc on base64:
> the caller must Close the returned encoder to flush any partially written blocks

If the previous blocks are wrong, I guess we don't care much about the end of the buff. The previous bytes are *already* written, it's only for for *end* of the buffer, if the size is not a multiple of 3, that the encoder has to add some padding bytes (see Base64 encoding on Wiki, very well explained ;)), that's why we must call `Close`. But if the rest is faulty, we don't *have* to call `Close`